core-web/libs/edit-content/src/lib/fields/dot-edit-content-custom-field/dot-edit-content-custom-field.component.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { Spectator, createComponentFactory, mockProvider } from '@ngneat/spectator/jest';
+
+import { ControlContainer, FormControl, FormGroup, FormGroupDirective } from '@angular/forms';
+
+import { DotContentTypeService, DotWorkflowActionsFireService } from '@dotcms/data-access';
+
+import { DotEditContentCustomFieldComponent } from './dot-edit-content-custom-field.component';
+
+import { DotEditContentService } from '../../services/dot-edit-content.service';
+import { CUSTOM_FIELD_MOCK, createFormGroupDirectiveMock } from '../../utils/mocks';
+
+describe('DotEditContentCustomFieldComponent', () => {
+    let spectator: Spectator<DotEditContentCustomFieldComponent>;
+
+    const FAKE_FORM_GROUP = new FormGroup({
+        custom: new FormControl()
+    });
+
+    const createComponent = createComponentFactory({
+        component: DotEditContentCustomFieldComponent,
+        detectChanges: false,
+        componentViewProviders: [
+            { provide: ControlContainer, useValue: createFormGroupDirectiveMock(FAKE_FORM_GROUP) }
+        ],
+        providers: [
+            FormGroupDirective,
+            {
+                provide: DotEditContentService,
+                useValue: {
+                    currentContentType: 'test'
+                }
+            },
+            mockProvider(DotContentTypeService),
+            mockProvider(DotWorkflowActionsFireService)
```

Comment:
I just check, is because the `DotEditContentService` uses `DotWorkflowActionsFireService` and you are not mocking `DotEditContentService`.

But now the questions here is bigger, why are you bringing all these services to the test if at first look the component you are testing here is no use to any of those, I think.

Also, I just noticed that this component is importing `BlockEditorModule` and `DotEditContentBinaryFieldComponent`, should those not be included in the `DotEditContentFieldsModule` as well?

Reply:
This test all fields inside, so we need to create a "TestBed" to support all fields, thats why we need to mock that providers
I dont like at 100% this, because is too coupled, but it works for now to have a generic tests for all fields

We can change the internal test to only test certain components, like the "simple" components
And test the "complex" components separately, on its own spec file, maybe?
cc: @rjvelazco @zJaaal

Reply:
We should talk about this in the next meeting.